### PR TITLE
[#9657] system messages ui changes in group chat

### LINKED ISSUE #9657: Correct the appearance of group chat system messages
# Problem

Currently system messages in group chats appear like regular messages with mentions inside. It's an improvement over the previous UI, but that's not at all how we want them to look like.

## Implementation

Screenshot 2019-12-12 at 16 12 51

---

each system message is just text centred in the chat thread. 8px margin on all sides. Black for username, Grey for the rest.

Group 44

---

and that's how they appear in a chat

Group Chat -_ Someone left chat

Usernames are not links, anonymous random names appear without the `@` character preceding them.

## Acceptance Criteria

Implemented according to design. [Link to Figma here](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=480%3A361)

## Notes

Thank you!

### PR DESCRIPTION
**Pull request by ghmulti**
Fixes #9657

### Summary

Suggest to create separate task for joining together events without separation (chained events) as this pr affects mostly styling 

#### Platforms
Tested on Android 

#### Areas that maybe impacted
Group chats

status: wip (awaiting feedback)

Screenshot from 2020-01-12 20-23-20

Colored block for margin/padding overview
Screenshot from 2020-01-12 20-21-24

### CODE REVIEW

**Comment by errorists** [comment]
IMG_1260
getting this error when trying to connect with mail servers, unfortunately can't test it out

looking at the screenshots, it seems like the horizontal margin padding on those messages is too large, should be 8.

**Comment by flexsurfer** [comment]
mailserver error introduced not in this PR

**Comment by errorists** [comment]
@flexsurfer totally, just stops me from testing in as I can't setup a group chat between a couple devices

**Comment by errorists** [comment]
@ghmulti ok managed to test it, these look correct as in the design apart from the horizontal margin which should be 8px on both left and right sides, so it's even with chat bubbles.

**Review by flexsurfer — DISMISSED** [review]

**Comment by ghmulti** [comment]
@errorists sure, there is a max-width limit defined for chat message view (red rectangle), should it be dropped for system messages then?
`:max-width      (if platform/desktop? 500 320)`

here is view after removing max-width for system messages and adding margin 8px on both sides
Screenshot from 2020-01-13 20-27-58
Screenshot from 2020-01-13 20-29-41

**Comment by ghmulti** [comment]
it's worth mentioning that changes related to "system messages" are made within generic chat message component, so style changes potentially may affect public groups and one-to-one chat views

**Comment by errorists** [comment]
@ghmulti thanks, that did the trick! I asked @Ferossgp to review changes in message code since he's too working on them and addressed some of the values in his PR here https://github.com/status-im/status-react/pull/9757

**Comment by Ferossgp** [comment]
Hey @ghmulti, good work! I just have some suggestions, that may have made this type of message less dependent on the text message. What do you think about making them as a separate type in multi-method `message-content`. That way you will have to check `system-message?` only in one place and will avoid passing that value through components. You can do that check one-time `chat-message`.

Thanks!

**Comment by churik** [comment]
Hello @ghmulti!
Thank you for contribution.
Can you please resolve conflicts and rebase your branch to current `develop`?

**Comment by ghmulti** [comment]
@Ferossgp hey, thanks for feedback and I think you are right - had the same thought

@churik sure, but I'll need to do some refactoring (https://github.com/status-im/status-react/pull/9800#issuecomment-574067629)

**Comment by churik** [comment]
I meant when it will be ready. Thanks!

**Comment by ghmulti** [comment]
for now - since there is no content-type-system-text defined in status-go, it is assigned manually based on status-message? flag in chat-message
`(when system-message? {:content-type constants/content-type-system-text})`
(if this is permissible) after adding new content-type to status-go, that verification needs to be deleted

**Review by errorists — APPROVED** [review]
Tested latest commit, group chat system messages look as they should now, thanks!!

File: src/status_im/constants.cljs
Comment (flexsurfer) [inline src/status_im/constants.cljs:15]: @cammellos could you take a look please?

File: src/status_im/ui/screens/chat/message/message.cljs
Comment (cammellos) [inline src/status_im/ui/screens/chat/message/message.cljs]: This is probably not the way to do it, if we want to have a `constansts/content-type-system-text` we should just generate this message with the right content-type (in status-go), so we avoid having to add it at runtime.

**Review by cammellos — CHANGES_REQUESTED** [review]
We should probably add the system type in status-go, I can take a look at it if you'd like me to do it

**Comment by cammellos** [comment]
@ghmulti You can cherry pick this commit https://github.com/status-im/status-react/commit/67372399040f6ea13478ba4dc4b35bfc8172030a , or use this branch (I have rebased against latest develop) https://github.com/status-im/status-react/compare/feature/group-chat-system-message-styling?expand=1 

Group chats messages will have the expected content-type. 

Please do not merge if tested successfully as we still need to merge the changes in status-go, once is tested ok, I can merge those and we can merge this PR.

**Comment by ghmulti** [comment]
@cammellos done :+1:

File: src/status_im/ui/screens/chat/message/message.cljs
Comment (cammellos) [inline src/status_im/ui/screens/chat/message/message.cljs]: you can change this to make it a bit neater I think, without using the merge that also might not be very performing, as this is a critical part of the code, something like:

```
{:style {:color (cond....)}
  :on-press (when-not (=...))} 
```
as if `on-press` is  `nil` it should be fine

**Review by cammellos — APPROVED** [review]
@Serhy could you give it another quick round of testing? if this tests ok I can merge status-go and update the version with develop.

File: src/status_im/ui/screens/chat/message/message.cljs
Comment (ghmulti) [reply src/status_im/ui/screens/chat/message/message.cljs]: done

**Comment by churik** [comment]
@ghmulti sorry for the significant delay.
Can you please rebase your branch with the latest `develop`, because builds can't be built now?

**Comment by ghmulti** [comment]
@cammellos  could you please prepare new status-go build with content-type, assume it will not work with the old build after rebase

**Comment by cammellos** [comment]
@ghmulti sorry for the delay, I have been off for the past two weeks.
You can use this branch:
`feature/add-content-type-group-chat`

To update status-go you can run:

`make shell`
`scripts/update-status-go.sh 'feature/add-content-type-group-chat'`

and then you can restart the stack and it should pick up the new version.

**Comment by ghmulti** [comment]
@churik hey, pushed updated version could you take a look please?
as @cammellos said if everything looks fine he will need to merge status-go and update the version with develop before merging pr

**Comment by churik** [comment]
Tested ally types of group system messages, right order, fetching system messages from offline (Android 10, IOS 13).
Group tests should work with, however can't check because it is separate repo.

Thanks for a work @ghmulti !

**Comment by flexsurfer** [comment]
https://github.com/status-im/status-react/pull/10208

**Pull request closed without merging**
